[10.0][CHG] invader_payment_adyen

rousseldenis opened the pull request:
![image](https://user-images.githubusercontent.com/19529533/67020136-02274c00-f0fe-11e9-972b-25014d0cae5c.png)

rousseldenis committed:
[WIP][10.0][CHG] invader_payment_adyen

rousseldenis committed:
[WIP]

rousseldenis commented:
@sbidoul @Cedric-Pigeon

Cedric-Pigeon commented on invader_payment_adyen/models/payment_transaction.py:
```diff
@@ -0,0 +1,11 @@
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    adyen_payment_data = fields.Char(groups="base.group_user")
```
I do not catch the use of "groups" attribute

rousseldenis replied:
I replicated Odoo standard fields attributes

Cedric-Pigeon commented on invader_payment_adyen/services/payment_adyen.py:
```diff
@@ -0,0 +1,425 @@
+
+    def _get_formatted_amount(self, currency, amount):
+        """
+        The expected amount format by Stripe
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/README.rst:
```diff
@@ -1,6 +1,6 @@
-Shopinvader Payment Adyen
-=========================
+===========================================
+Stripe Payment Acquirer (REST, Shopinvader)
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/README.rst:
```diff
@@ -19,7 +19,7 @@ Shopinvader Payment Adyen
 |badge1| |badge2| |badge3| 
 
-Implements Adyen payment
+REST Services for Stripe Payments (/shopinvader route).
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/readme/DESCRIPTION.rst:
```diff
@@ -1 +1 @@
-Implements Adyen payment
+REST Services for Stripe Payments (/shopinvader route).
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/static/description/index.html:
```diff
@@ -4,7 +4,7 @@
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
 <meta name="generator" content="Docutils 0.15.1: http://docutils.sourceforge.net/" />
-<title>Shopinvader Payment Adyen</title>
+<title>Stripe Payment Acquirer (REST, Shopinvader)</title>
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/static/description/index.html:
```diff
@@ -360,15 +360,15 @@
 <body>
-<div class="document" id="shopinvader-payment-adyen">
-<h1 class="title">Shopinvader Payment Adyen</h1>
+<div class="document" id="stripe-payment-acquirer-rest-shopinvader">
```
Adyen

Cedric-Pigeon commented on shopinvader_payment_adyen/static/description/index.html:
```diff
@@ -360,15 +360,15 @@
-<p><a class="reference external" href="https://odoo-community.org/page/development-status"><img alt="Beta" src="https://img.shields.io/badge/maturity-Beta-yellow.png" /></a> <a class="reference external" href="http://www.gnu.org/licenses/agpl-3.0-standalone.html"><img alt="License: AGPL-3" src="https://img.shields.io/badge/licence-AGPL--3-blue.png" /></a> <a class="reference external" href="https ...
-<p>Implements Adyen payment</p>
+<p><a class="reference external" href="https://odoo-community.org/page/development-status"><img alt="Beta" src="https://img.shields.io/badge/maturity-Beta-yellow.png" /></a> <a class="reference external" href="http://www.gnu.org/licenses/agpl-3.0-standalone.html"><img alt="License: AGPL-3" src="https://img.shields.io/badge/licence-AGPL--3-blue.png" /></a> <a class="reference external" href="https ...
+<p>REST Services for Stripe Payments (/shopinvader route).</p>
```
Adyen

rousseldenis committed:
[FIX] shopinvader_payment_adyen: Change strings

rousseldenis committed:
[10.0][IMP] invader_payment_adyen: Set correctly acquirer reference + log

lmignon reviewed (approved):
@rousseldenis  Can you squash your commit 'WIP' PLZ

rousseldenis committed:
[10.0][IMP] invader_payment_adyen: Set live prefix

rousseldenis commented:
I've added webhook notifications here : https://github.com/acsone/odoo-shopinvader-payment/pull/10

@lmignon @sbidoul @Cedric-Pigeon

rousseldenis committed:
[10.0][IMP] invader_paypment_adyen: Add paymentResult as GET

rousseldenis committed:
[10.0][IMP] invader_payment_adyen: Manage Received status

As the Received status is an intermediate state that means Adyen received
payment but waits for bank acknowledgment. This behaviour needs to be
the same as manual payments (session invalidation and cart confirmation -
typology change).

rousseldenis committed:
[10.0][IMP] shop/invader_payment_adyen: Add tests for Received status

lmignon commented:
/ocabot merge nobump
